### PR TITLE
gitlab-ci-multi-runner: add service plist

### PR DESCRIPTION
**Pull request by michalkosmowski**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Adds plist to gitlab-ci-multi-runner formula.

**Q:** Is there any better way of injecting user's home path?
**Q:** Any tips to make an alias for this service? `brew services start gitlab-ci-multi-runner` is really long.

### CODE REVIEW

File: Formula/gitlab-ci-multi-runner.rb [hunk 1]
```diff
@@ -75,6 +75,36 @@ def install
     end
   end
 
+  plist_options :startup => true, :manual => "gitlab-ci-multi-runner start"
```
Comment (MikeMcQuaid) [inline Formula/gitlab-ci-multi-runner.rb]: Drop the `:startup => true`

File: Formula/gitlab-ci-multi-runner.rb [hunk 2]
```diff
@@ -75,6 +75,36 @@ def install
...
+        <key>RunAtLoad</key><true/>
+        <key>Disabled</key><false/>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_bin}/gitlab-ci-multi-runner</string>
+          <string>run</string>
+          <string>--working-directory</string>
+          <string>#{File.expand_path("~")}</string>
```
Comment (MikeMcQuaid) [inline Formula/gitlab-ci-multi-runner.rb]: It may just be worth using `ENV["HOME"]`

File: Formula/gitlab-ci-multi-runner.rb [hunk 3]
```diff
@@ -75,6 +75,36 @@ def install
...
+          <string>--config</string>
+          <string>#{File.expand_path("~")}/.gitlab-runner/config.toml</string>
```
Comment (MikeMcQuaid) [inline Formula/gitlab-ci-multi-runner.rb]: What if this file doesn't exist?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/gitlab-ci-multi-runner.rb (hunk 3 above)
Comment (michalkosmowski) [inline Formula/gitlab-ci-multi-runner.rb]: The runner works without the config file, but cannot connect to Gitlab. It retries to load the config file every ~1s (based on the logs). 

That config file is a result of `gitlab-ci-multi-runner register` where the user passes a token, so I think we should be good here.

Comment (MikeMcQuaid) [inline Formula/gitlab-ci-multi-runner.rb]: We'll need to figure out a way to make it clearer that this requires configuration before the plist can be used. To me that makes it a bad plist candidate, personally.

Comment (michalkosmowski) [inline Formula/gitlab-ci-multi-runner.rb]: I see your point. On the other hand I'm not sure if anyone will miss their docs about it:
https://docs.gitlab.com/ce/ci/runners/README.html#creating-and-registering-a-runner
Technically the plist can be used, the service works on your machine, it just doesn't connect further to gitlab.com.

Anyway, i'm just scratching my itch 😉

**Comment by michalkosmowski** [comment]
Hey @MikeMcQuaid any chances to merge this in?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**